sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedAsyncClient.java

```diff
@@ -0,0 +1,111 @@
+public class BlobChangefeedAsyncClient {
+
+    static final String CHANGEFEED_CONTAINER_NAME = "$blobchangefeed";
+
+    private final BlobContainerAsyncClient client;
+    private final BlobChangefeedPagedFluxFactory pagedFluxFactory;
+
+    /**
+     * Package-private constructor for use by {@link BlobChangefeedClientBuilder}.
+     *
+     * @param pipeline The pipeline used to send and receive service requests.
+     * @param url The endpoint where to send service requests.
+     * @param version The version of the service to receive requests.
+     */
+    BlobChangefeedAsyncClient(HttpPipeline pipeline, String url, BlobServiceVersion version) {
+        this.client = new BlobContainerClientBuilder()
+            .endpoint(url)
+            .containerName(CHANGEFEED_CONTAINER_NAME)
+            .pipeline(pipeline)
+            .serviceVersion(version)
+            .buildAsyncClient();
+        AvroReaderFactory avroReaderFactory = new AvroReaderFactory();
+        BlobLazyDownloaderFactory blobLazyDownloaderFactory = new BlobLazyDownloaderFactory();
+        ChunkFactory chunkFactory = new ChunkFactory(avroReaderFactory, blobLazyDownloaderFactory, client);
+        ShardFactory shardFactory = new ShardFactory(chunkFactory, client);
+        SegmentFactory segmentFactory = new SegmentFactory(shardFactory, client);
+        ChangefeedFactory changefeedFactory = new ChangefeedFactory(segmentFactory, client);
+        this.pagedFluxFactory = new BlobChangefeedPagedFluxFactory(changefeedFactory);
+    }
+
+    /**
+     * Returns a reactive Publisher emitting all the changefeed events for this account lazily as needed.
+     *
+     * <p>
+     * Changefeed events are returned in approximate temporal order.
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-change-feed?tabs=azure-portal">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.changefeed.BlobChangefeedAsyncClient.getEvents}
+     *
+     * @return A reactive response emitting the changefeed events.
+     */
+    public BlobChangefeedPagedFlux getEvents() {
+        return getEvents(null, null);
+    }
+
+    /**
+     * Returns a reactive Publisher emitting all the changefeed events for this account lazily as needed.
+     *
+     * <p>
+     * Changefeed events are returned in approximate temporal order.
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-change-feed?tabs=azure-portal">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.changefeed.BlobChangefeedAsyncClient.getEvents#OffsetDateTime-OffsetDateTime}
+     *
+     * @param startTime Filters the results to return events approximately after the start time.
```

Comment:
Should we give a little more detail about how approximate this is?

Reply:
added detail